src/aggregate/aggregate_request.c

```diff
@@ -1412,36 +1410,32 @@ int AREQ_BuildPipeline(AREQ *req, int options, QueryError *status) {
         }
         // Get all the keys for this lookup...
         while (!AC_IsAtEnd(&lstp->args)) {
-          const char *path = AC_GetStringNC(&lstp->args, NULL);
+          size_t name_len;
+          const char *name, *path = AC_GetStringNC(&lstp->args, &name_len);
           if (*path == '@') {
             path++;
+            name_len--;
           }
-          const char *name = path;
-
-          RLookupKey *kk = RLookup_GetKey_TEMP(curLookup, path, RLOOKUP_F_OEXCL | RLOOKUP_F_OCREAT);
-          if (!kk) {
-            // We only get a NULL return if the key already exists, which means
-            // that we don't need to retrieve it again.
-            continue;
-          }
-
           if (AC_AdvanceIfMatch(&lstp->args, SPEC_AS_STR)) {
-            int rv = AC_GetString(&lstp->args, &name, NULL, 0);
+            int rv = AC_GetString(&lstp->args, &name, &name_len, 0);
             if (rv != AC_OK) {
               QERR_MKBADARGS_FMT(status, "LOAD path AS name - must be accompanied with NAME");
               return REDISMODULE_ERR;
             } else if (!strcasecmp(name, SPEC_AS_STR)) {
               QERR_MKBADARGS_FMT(status, "Alias for LOAD cannot be `AS`");
               return REDISMODULE_ERR;
             }
+          } else {
+            // Set the name to the path. name_len is already the length of the path.
+            name = path;
           }
-          // set lookupkey name to name.
-          // by defualt "name = path"
-          kk->name = name;
-          kk->name_len = strlen(name);
 
-          kk->flags |= RLOOKUP_F_ISLOADED;
-          lstp->keys[lstp->nkeys++] = kk;
+          RLookupKey *kk = RLookup_GetKey_LoadEx(curLookup, name, name_len, path, loadFlags);
+          // We only get a NULL return if the key already exists, which means
+          // that we don't need to retrieve it again.
+          if (kk) {
```

Comment:
Confirming
Before if `RLookup_GetKey_TEMP` returned null there was a `continue`, but now it is not needed since it is at the end of the loop block?

Reply:
We previously first created a key by the path, and then continued to check if there is an alias.
Since I moved the key creation and validation to the end, I reversed the condition check